ext/strings_test.go

```diff
@@ -1254,36 +1247,38 @@ func TestStringFormat(t *testing.T) {
 		}
 		return opts
 	}
-	buildOpts := func(locale string, variables []cel.EnvOption) []cel.EnvOption {
+	buildOpts := func(skipCompileCheck bool, locale string, variables []cel.EnvOption) []cel.EnvOption {
 		opts := []cel.EnvOption{
-			Strings(StringsLocale(locale)),
+			Strings(StringsLocale(locale), StringsValidateFormatCalls(!skipCompileCheck)),
 			cel.Container("ext"),
 			cel.Abbrevs("google.expr.proto3.test"),
 			cel.Types(&proto3pb.TestAllTypes{}),
-			cel.ASTValidators(cel.ValidateHomogeneousAggregateLiterals()),
 			NativeTypes(
 				reflect.TypeOf(&TestNestedType{}),
 				reflect.ValueOf(&TestAllTypes{}),
 			),
 		}
+		if !skipCompileCheck {
+			opts = append(opts, cel.ASTValidators(cel.ValidateHomogeneousAggregateLiterals()))
+		}
 		opts = append(opts, variables...)
 		return opts
 	}
 	runCase := func(format, formatArgs, locale string, dynArgs map[string]any, skipCompileCheck bool, expectedRuntimeCost uint64, expectedEstimatedCost checker.CostEstimate, t *testing.T) (ref.Val, error) {
-		env, err := cel.NewEnv(buildOpts(locale, buildVariables(dynArgs))...)
+		env, err := cel.NewEnv(buildOpts(skipCompileCheck, locale, buildVariables(dynArgs))...)
 		if err != nil {
 			t.Fatalf("cel.NewEnv() failed: %v", err)
 		}
 		expr := fmt.Sprintf("%q.format([%s])", format, formatArgs)
 		if len(dynArgs) == 0 {
-			return evalExpr(env, expr, cel.NoVars(), skipCompileCheck, expectedRuntimeCost, expectedEstimatedCost, t)
+			return evalExpr(env, expr, cel.NoVars(), expectedRuntimeCost, expectedEstimatedCost, t)
 		}
-		return evalExpr(env, expr, dynArgs, skipCompileCheck, expectedRuntimeCost, expectedEstimatedCost, t)
+		return evalExpr(env, expr, dynArgs, expectedRuntimeCost, expectedEstimatedCost, t)
 	}
 	checkCase := func(output ref.Val, expectedOutput string, err error, expectedErr string, t *testing.T) {
 		if err != nil {
 			if expectedErr != "" {
-				if err.Error() != expectedErr {
+				if !strings.Contains(err.Error(), expectedErr) {
 					t.Fatalf("expected %q as error message, got %q", expectedErr, err.Error())
```

Comment:
Super tiny nit: I think the `strings.Contains` change is good, though maybe it might be better for this line to now read something like `"expected error message to contain %q, got %q"` as a result.